src/Response/Types/ErrorType.php

```diff
@@ -5,30 +5,166 @@
+    {
+        $headers = $response->getHeaders();
+        $headerKeys = $this->getHeaderKeys($headers);
+
+        for ($x = 0; $x < count($headerKeys); $x++) {
+            $errorDescription = $this->addPlaceHolderValue(
+                $errorDescription,
+                '{$response.header.' . $headerKeys[$x] . '}',
+                $headers[$headerKeys[$x]]
+            );
+        }
+
+        return $errorDescription;
+    }
+
+    private function updateResponsePlaceholderValues($errorDescription, $jsonPointersInTemplate, $response)
+    {
+        if (count($jsonPointersInTemplate[0]) > 0) {
+            $jsonResponsePointer = $this->initializeJsonPointer($response);
+
+            $jsonPointers = $jsonPointersInTemplate[0];
+
+            for ($x = 0; $x < count($jsonPointers); $x++) {
+                $placeHolderValue = "";
+
+                if ($jsonResponsePointer != null) {
+                    $placeHolderValue = $this->getJsonPointerValue($jsonResponsePointer, ltrim($jsonPointers[$x], '#'));
+                }
+
+                $errorDescription = $this->addPlaceHolderValue(
+                    $errorDescription,
+                    '{$response.body' . $jsonPointers[$x] . '}',
+                    $placeHolderValue
+                );
+            }
+
+            return $errorDescription;
+        }
+
+        return $this->addPlaceHolderValue(
+            $errorDescription,
+            '{$response.body}',
+            $response->getRawBody()
+        );
+    }
+
+    private function getJsonPointersFromTemplate($template)
+    {
+        $pointerPattern = '/#[\w\/]*/i';
+
+        preg_match_all($pointerPattern, $template, $matches);
+
+        return $matches;
+    }
+
+    private function addPlaceHolderValue($template, $placeHolder, $value)
+    {
+        if (!is_string($value)) {
+            $value = var_export($value, true);
+        }
+
+        return str_replace($placeHolder, $value, $template);
+    }
+
+    private function getHeaderKeys($headers)
+    {
+        return array_keys($headers);
+    }
+
+    private function getJsonPointerValue($jsonPointer, $pointer)
+    {
+        try {
+            if (trim($pointer) === '') {
+                return "";
+            }
+
+            $pointerValue = $jsonPointer->get($pointer);
+
+            if (is_object($pointerValue)) {
+                return CoreHelper::serialize($pointerValue);
+            }
+
+            return $pointerValue;
+        } catch (Pointer\NonexistentValueReferencedException $ex) {
+            return "";
+        }
+    }
+
+    private function initializeJsonPointer($response)
+    {
+        $rawBody = $response->getRawBody();
+        $jsonResponsePointer = null;
+
+        if ($this->isJson($rawBody)) {
+            $jsonResponsePointer = new Pointer($rawBody);
+        }
+
+        return $jsonResponsePointer;
+    }
+
+    private function isJson($string)
+    {
+        json_decode($string);
```

Comment:
Whats the reason for not using CoreHelper > deserialize, then checking the value if its an array or object to confirm `isJson`